Controller/Admin/ModuleSupport.php

```diff
@@ -0,0 +1,195 @@
+        }
+
+        return $this->_sThisTemplate;
+    }
+
+    /**
+     * Action triggered from form to send support email
+     *
+     * @return void
+     */
+    public function sendSupportEmailAction()
+    {
+        try {
+            $this->_validateRequest();
+        } catch (Exception $e) {
+            $this->_aViewData['alertMessage'] = $e->getMessage();
+            $this->_aViewData['alertType'] = 'error';
+            return;
+        }
+
+        $aEmailData = [];
+
+        $this->_addDataFromForm($aEmailData);
+        $this->_addShopData($aEmailData);
+
+        $this->_sendEmail($aEmailData);
+    }
+
+    /**
+     * Creates the email object and tries to send it
+     *
+     * @param array $aEmailData
+     */
+    protected function _sendEmail($aEmailData)
+    {
+
+        $oEmail = oxNew(\Wirecard\Oxid\Extend\Email::class);
+
+        if ($oEmail->sendSupportEmail($aEmailData)) {
+            $this->_aViewData['alertMessage'] = Helper::translate('success_email');
+            $this->_aViewData['alertType'] = 'success';
+            $this->_aViewData['replyToEmail'] = '';
+            $this->_aViewData['fromEmail'] = '';
+            $this->_aViewData['body'] = '';
+        } else {
+            $this->_aViewData['alertMessage'] = Helper::translate('support_send_error');
+            $this->_aViewData['alertType'] = 'error';
+        }
+    }
+
+    /**
+     * Adds data received from form into the supplied array
+     *
+     * @param array $aEmailData
+     */
+    protected function _addDataFromForm(&$aEmailData)
+    {
+        $body = $this->getConfig()->getRequestParameter('module_support_text');
+        $sFromEmail = $this->getConfig()->getRequestParameter('module_support_email_from');
+        $sReplyToEmail = $this->getConfig()->getRequestParameter('module_support_email_reply');
+
+        $aEmailData = array_merge($aEmailData, [
+            'body' => $body,
+            'replyTo' => $sReplyToEmail,
+            'from' => $sFromEmail
+        ]);
+    }
+
+    /**
+     * Adds automatically provided data into the supplied array
+     *
+     * @param array $aEmailData email data array
+     */
+    protected function _addShopData(&$aEmailData)
+    {
+        $aEmailData = array_merge($aEmailData, [
+            'modules' => $this->_getOtherModules(),
+            'module' => $this->_getModule(),
+            'shopVersion' => $this->getConfig()->getVersion(),
+            'shopEdition' => $this->getConfig()->getFullEdition(),
+            'phpVersion' => phpversion(),
+            'system' => php_uname(),
+            'subject' => Helper::translate('support_email_subject'),
+            'recipient' => $this->_getModule()->getInfo('email'),
+            'payments' => Helper::getModulePaymentsIncludingInactive()
+        ]);
+    }
+
+    /**
+     * Returns current module
+     *
+     * @return OxidEsales\Eshop\Core\Module\Module
+     */
+    protected function _getModule()
```

Comment:
please update the comment, returns current module i still have no idea what in detail is it doing.